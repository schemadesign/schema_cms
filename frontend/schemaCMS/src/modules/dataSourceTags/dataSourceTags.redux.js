import { createReducer } from 'reduxsauce';
import Immutable from 'seamless-immutable';
import { createRoutine } from 'redux-saga-routines';

const PREFIX = 'DATA_SOURCE_TAGS/';

export const DataSourceTagsRoutines = {
  fetchDataSourceTags: createRoutine(`${PREFIX}_FETCH_DATA_SOURCE_TAGS`),
  updateDataSourceTags: createRoutine(`${PREFIX}_UPDATE_DATA_SOURCE_TAGS`),
};

export const INITIAL_STATE = new Immutable({
  dataSourceTags: [],
});

const setDataSourceTags = (state = INITIAL_STATE, { payload }) => state.set('dataSourceTags', payload);

export const reducer = createReducer(INITIAL_STATE, {
  [DataSourceTagsRoutines.fetchDataSourceTags.SUCCESS]: setDataSourceTags,
  [DataSourceTagsRoutines.updateDataSourceTags.SUCCESS]: setDataSourceTags,
});
