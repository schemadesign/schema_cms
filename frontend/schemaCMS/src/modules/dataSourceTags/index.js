export { DataSourceTagsRoutines } from './dataSourceTags.redux';

export * from './dataSourceTags.selectors';
