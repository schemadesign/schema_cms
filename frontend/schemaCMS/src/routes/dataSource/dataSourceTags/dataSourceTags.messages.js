/* eslint-disable max-len */
import { defineMessages } from 'react-intl';

export default defineMessages({
  subTitle: {
    id: 'dataSource.dataSourceTags.subTitle',
    defaultMessage: 'Tags',
  },
  save: {
    id: 'dataSource.dataSourceTags.save',
    defaultMessage: 'Save',
  },
});
