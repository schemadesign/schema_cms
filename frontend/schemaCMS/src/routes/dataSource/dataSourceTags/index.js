export { default as DataSourceTags } from './dataSourceTags.container';
