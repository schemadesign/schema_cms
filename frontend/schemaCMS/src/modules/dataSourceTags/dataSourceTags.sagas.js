import { all, put, takeLatest } from 'redux-saga/effects';
import reportError from '../../shared/utils/reportError';

import { DataSourceTagsRoutines } from './dataSourceTags.redux';
import api from '../../shared/services/api';
import { DATA_SOURCES_PATH, TAGS_PATH } from '../../shared/utils/api.constants';

function* fetchDataSourceTags({ payload }) {
  try {
    yield put(DataSourceTagsRoutines.fetchDataSourceTags.request());
    const { data } = yield api.get(`${DATA_SOURCES_PATH}/${payload.dataSourceId}${TAGS_PATH}`);

    yield put(DataSourceTagsRoutines.fetchDataSourceTags.success(data.results));
  } catch (error) {
    reportError(error);
    yield put(DataSourceTagsRoutines.fetchDataSourceTags.failure(error));
  } finally {
    yield put(DataSourceTagsRoutines.fetchDataSourceTags.fulfill());
  }
}

function* updateDataSourceTags({ payload: { dataSourceId, formData } }) {
  try {
    yield put(DataSourceTagsRoutines.updateDataSourceTags.request());
    const { data } = yield api.patch(`${DATA_SOURCES_PATH}/${dataSourceId}${TAGS_PATH}`, formData);

    yield put(DataSourceTagsRoutines.updateDataSourceTags.success(data.results));
  } catch (error) {
    reportError(error);
    yield put(DataSourceTagsRoutines.updateDataSourceTags.failure(error));
  } finally {
    yield put(DataSourceTagsRoutines.updateDataSourceTags.fulfill());
  }
}

export function* watchDataSourceTags() {
  yield all([
    takeLatest(DataSourceTagsRoutines.fetchDataSourceTags.TRIGGER, fetchDataSourceTags),
    takeLatest(DataSourceTagsRoutines.updateDataSourceTags.TRIGGER, updateDataSourceTags),
  ]);
}
