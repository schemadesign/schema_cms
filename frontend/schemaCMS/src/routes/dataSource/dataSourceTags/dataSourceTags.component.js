import React, { useState } from 'react';
import { FormattedMessage, useIntl } from 'react-intl';
import { useEffectOnce } from 'react-use';
import PropTypes from 'prop-types';
import { useFormik } from 'formik';
import { useHistory, useRouteMatch } from 'react-router';
import Helmet from 'react-helmet';
import { defaultTo, flatten, groupBy, keys, map, pipe, prop } from 'ramda';

import { Container } from './dataSourceTags.styles';
import messages from './dataSourceTags.messages';
import reportError from '../../../shared/utils/reportError';
import { TagSearch } from '../../../shared/components/tagSearch';
import { LoadingWrapper } from '../../../shared/components/loadingWrapper';
import { MobileMenu } from '../../../shared/components/menu/mobileMenu';
import { errorMessageParser, filterMenuOptions } from '../../../shared/utils/helpers';
import { ContextHeader } from '../../../shared/components/contextHeader';
import { DataSourceNavigation } from '../../../shared/components/dataSourceNavigation';
import { getProjectMenuOptions } from '../../project/project.constants';
import { NavigationContainer, NextButton } from '../../../shared/components/navigation';

export const DataSourceTags = ({
  fetchTagCategories,
  fetchDataSourceTags,
  updateDataSourceTags,
  project,
  tagCategories,
  dataSource,
  userRole,
  tags,
}) => {
  const [error, setError] = useState(null);
  const [loading, setLoading] = useState(true);
  const intl = useIntl();
  const history = useHistory();
  const match = useRouteMatch();
  const initialValues = pipe(
    defaultTo([]),
    map(item => ({ ...item, label: item.value })),
    groupBy(prop('category'))
  )(tags);
  const { values, setFieldValue, handleSubmit, isSubmitting, dirty } = useFormik({
    initialValues,
    onSubmit: async (data, { setSubmitting, setErrors }) => {
      try {
        const formData = pipe(
          keys,
          map(key => map(({ value }) => ({ category: parseInt(key, 10), value }))(data[key])),
          flatten
        )(data);
        await updateDataSourceTags({ dataSourceId: dataSource.id, formData });
        setSubmitting(false);
      } catch (errors) {
        const { formatMessage } = intl;
        const errorMessages = errorMessageParser({ errors, messages, formatMessage });

        setErrors(errorMessages);
      }
    },
  });
  const headerTitle = dataSource.name;
  const headerSubtitle = <FormattedMessage {...messages.subTitle} />;
  const menuOptions = getProjectMenuOptions(project.id);

  useEffectOnce(() => {
    (async () => {
      try {
        const fetchTagCategoriesPromise = fetchTagCategories({ projectId: project.id, type: 'dataset' });
        const fetchDataSourceTagsPromise = fetchDataSourceTags({ dataSourceId: dataSource.id });
        await Promise.all([fetchDataSourceTagsPromise, fetchTagCategoriesPromise]);
      } catch (e) {
        reportError(e);
        setError(e);
      } finally {
        setLoading(false);
      }
    })();
  });

  return (
    <Container>
      <Helmet title={intl.formatMessage(messages.subTitle)} />
      <MobileMenu
        headerTitle={headerTitle}
        headerSubtitle={headerSubtitle}
        options={filterMenuOptions(menuOptions, userRole)}
      />
      <ContextHeader title={headerTitle} subtitle={headerSubtitle}>
        <DataSourceNavigation history={history} match={match} dataSource={dataSource} />
      </ContextHeader>
      <LoadingWrapper loading={loading} error={error}>
        <TagSearch tagCategories={tagCategories} values={values} setFieldValue={setFieldValue} />
        <NavigationContainer right fixed padding="10px 0 70px">
          <NextButton onClick={handleSubmit} loading={isSubmitting} disabled={!dirty || isSubmitting}>
            <FormattedMessage {...messages.save} />
          </NextButton>
        </NavigationContainer>
      </LoadingWrapper>
    </Container>
  );
};

DataSourceTags.propTypes = {
  fetchTagCategories: PropTypes.func.isRequired,
  fetchDataSourceTags: PropTypes.func.isRequired,
  updateDataSourceTags: PropTypes.func.isRequired,
  project: PropTypes.object.isRequired,
  dataSource: PropTypes.object.isRequired,
  tagCategories: PropTypes.object.isRequired,
  userRole: PropTypes.string.isRequired,
  tags: PropTypes.array,
};
