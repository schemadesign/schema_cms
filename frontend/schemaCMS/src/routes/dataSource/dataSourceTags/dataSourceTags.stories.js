import React from 'react';
import { storiesOf } from '@storybook/react';

import { DataSourceTags } from './dataSourceTags.component';
import { withTheme } from '../../../.storybook/decorators';
import { history, intl } from '../../../.storybook/helpers';
import { ROLES } from '../../../modules/userProfile/userProfile.constants';

export const defaultProps = {
  userRole: ROLES.ADMIN,
  dataSource: {
    project: { id: '1' },
  },
  tags: [
    {
      key: 'key 1',
      id: 1,
      isActive: true,
    },
    {
      key: 'key 2',
      id: 2,
      isActive: false,
    },
  ],
  fetchTags: Function.prototype,
  setTags: Function.prototype,
  history,
  intl,
  match: {
    params: {
      dataSourceId: '1',
    },
    url: '/datasource/1/tag',
  },
};

export const noTagsProps = {
  ...defaultProps,
  filters: [],
};

storiesOf('Data Source|DataSourceTags', module)
  .addDecorator(withTheme())
  .add('No Data', () => <DataSourceTags {...noTagsProps} />)
  .add('Default', () => <DataSourceTags {...defaultProps} />);
