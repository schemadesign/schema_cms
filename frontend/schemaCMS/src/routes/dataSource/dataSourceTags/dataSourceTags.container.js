import { connect } from 'react-redux';
import { bindPromiseCreators, promisifyRoutine } from 'redux-saga-routines';
import { createStructuredSelector } from 'reselect';
import { withRouter } from 'react-router-dom';
import { hot } from 'react-hot-loader';
import { compose } from 'ramda';

import { DataSourceTags } from './dataSourceTags.component';
import { selectTagCategories, TagCategoryRoutines } from '../../../modules/tagCategory';
import { selectProject } from '../../../modules/project';
import { selectDataSource } from '../../../modules/dataSource';
import { selectUserRole } from '../../../modules/userProfile';
import { DataSourceTagsRoutines } from '../../../modules/dataSourceTags';

const mapStateToProps = createStructuredSelector({
  tagCategories: selectTagCategories,
  project: selectProject,
  dataSource: selectDataSource,
  userRole: selectUserRole,
});

export const mapDispatchToProps = dispatch => ({
  ...bindPromiseCreators(
    {
      fetchTagCategories: promisifyRoutine(TagCategoryRoutines.fetchTagCategories),
      fetchDataSourceTags: promisifyRoutine(DataSourceTagsRoutines.fetchDataSourceTags),
      updateDataSourceTags: promisifyRoutine(DataSourceTagsRoutines.updateDataSourceTags),
    },
    dispatch
  ),
});

export default compose(
  hot(module),
  connect(
    mapStateToProps,
    mapDispatchToProps
  ),
  withRouter
)(DataSourceTags);
