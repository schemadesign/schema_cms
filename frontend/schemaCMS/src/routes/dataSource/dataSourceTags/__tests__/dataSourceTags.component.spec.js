import React from 'react';
import { shallow } from 'enzyme';
import { Formik } from 'formik';
import { Form } from 'schemaUI';

import { DataSourceTags } from '../dataSourceTags.component';
import { defaultProps, noTagsProps } from '../dataSourceTags.stories';

describe('DataSourceTags: Component', () => {
  const component = props => <DataSourceTags {...defaultProps} {...props} />;

  const render = (props = {}) => shallow(component(props));

  it('should render correctly with loader', () => {
    const wrapper = render();

    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly', async () => {
    const fetchTags = jest.fn().mockReturnValue(Promise.resolve());
    const wrapper = await render({
      fetchTags,
    });

    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly when no filters', async () => {
    const fetchTags = jest.fn().mockReturnValue(Promise.resolve());
    const wrapper = await render({
      ...noTagsProps,
      fetchTags,
    });

    expect(wrapper).toMatchSnapshot();
  });

  it('should call fetchTags on componentDidMount', () => {
    const fetchTags = jest.spyOn(defaultProps, 'fetchTags');

    render({
      fetchTags,
    });

    expect(fetchTags).toHaveBeenCalledTimes(1);
  });

  it('should call setTags on submit', async () => {
    const fetchTags = jest.fn().mockReturnValue(Promise.resolve());
    const setTags = jest.spyOn(defaultProps, 'setTags');
    const active = [];

    const wrapper = await render({
      fetchTags,
      setTags,
    }).find(Formik);

    wrapper
      .dive()
      .find(Form.CheckboxGroup)
      .simulate('change', { target: { checked: true } });

    wrapper.prop('onSubmit')(active, {
      setSubmitting: Function.prototype,
      setErrors: Function.prototype,
    });

    expect(setTags).toHaveBeenCalledTimes(1);
  });

  it('should set error correctly', async () => {
    const errorResponse = 'fetchTags should return error';
    const wrapper = await render({
      fetchTags: jest.fn().mockReturnValue(Promise.reject(errorResponse)),
    });

    const { loading, error } = wrapper.state();

    expect(loading).toBeFalsy();
    expect(error).toBe(errorResponse);
  });
});
