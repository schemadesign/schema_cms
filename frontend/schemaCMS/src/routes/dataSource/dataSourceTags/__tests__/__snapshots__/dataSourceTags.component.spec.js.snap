// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataSourceTags: Component should render correctly 1`] = `
<Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
    title="Tags List"
  />
  <MobileMenu
    headerSubtitle={
      <FormattedMessage
        defaultMessage="Tags List"
        id="dataSource.dataSourceTags.subTitle"
        values={Object {}}
      />
    }
    options={
      Array [
        Object {
          "allowedRoles": Array [
            "admin",
            "editor",
          ],
          "id": "projectsNavBtn",
          "label": <FormattedMessage
            defaultMessage="Projects"
            id="project.view.projectsListPage"
            values={Object {}}
          />,
          "to": "/project/",
          "type": "link_item",
        },
        Object {
          "allowedRoles": Array [
            "admin",
          ],
          "id": "usersPageNavBtn",
          "label": <FormattedMessage
            defaultMessage="Users"
            id="project.view.usersPage"
            values={Object {}}
          />,
          "to": "/user/",
          "type": "link_item",
        },
        Object {
          "allowedRoles": Array [
            "admin",
            "editor",
          ],
          "id": "projectDetailsNavBtn",
          "label": <FormattedMessage
            defaultMessage="Project details"
            id="project.view.projectDetails"
            values={Object {}}
          />,
          "to": "/project/1",
          "type": "link_item",
        },
        Object {
          "allowedRoles": Array [
            "admin",
            "editor",
          ],
          "id": "dataSourceNavBtn",
          "label": <FormattedMessage
            defaultMessage="Data Sources"
            id="project.view.projectDataSources"
            values={Object {}}
          />,
          "to": "/project/1/datasource",
          "type": "link_item",
        },
        Object {
          "allowedRoles": Array [
            "admin",
            "editor",
          ],
          "id": "stateNavBtn",
          "label": <FormattedMessage
            defaultMessage="States"
            id="project.view.states"
            values={Object {}}
          />,
          "to": "/project/1/state",
          "type": "link_item",
        },
        Object {
          "allowedRoles": Array [
            "admin",
          ],
          "id": "usersNavBtn",
          "label": <FormattedMessage
            defaultMessage="Project Users"
            id="project.view.projectUsersPage"
            values={Object {}}
          />,
          "to": "/project/1/user",
          "type": "link_item",
        },
        Object {
          "allowedRoles": Array [
            "admin",
            "editor",
          ],
          "id": "folderNavBtn",
          "label": <FormattedMessage
            defaultMessage="Pages"
            id="project.view.projectPages"
            values={Object {}}
          />,
          "to": "/project/1/folder",
          "type": "link_item",
        },
      ]
    }
  />
  <ContextHeader
    subtitle={
      <FormattedMessage
        defaultMessage="Tags List"
        id="dataSource.dataSourceTags.subTitle"
        values={Object {}}
      />
    }
  >
    <DataSourceNavigation
      dataSource={
        Object {
          "project": Object {
            "id": "1",
          },
        }
      }
      fetchTags={
        [MockFunction] {
          "calls": Array [
            Array [
              Object {
                "dataSourceId": "1",
              },
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": Promise {},
            },
          ],
        }
      }
      hideOnDesktop={false}
      history={
        Object {
          "goBack": [Function],
          "push": [Function],
        }
      }
      intl={
        Object {
          "formatMessage": [Function],
        }
      }
      match={
        Object {
          "params": Object {
            "dataSourceId": "1",
          },
          "url": "/datasource/1/tag",
        }
      }
      setTags={[Function]}
      tags={
        Array [
          Object {
            "id": 1,
            "isActive": true,
            "key": "key 1",
          },
          Object {
            "id": 2,
            "isActive": false,
            "key": "key 2",
          },
        ]
      }
      userRole="admin"
    />
  </ContextHeader>
  <LoadingWrapper
    error={null}
    loading={false}
    noData={false}
    noDataContent={null}
  >
    <styled.div>
      <styled.div>
        <Styled(Component)
          onClick={[Function]}
        >
          <Component />
        </Styled(Component)>
      </styled.div>
      <styled.div>
        <FormattedMessage
          defaultMessage="{tags, plural,one {# Tag List} other {# Tags List}}"
          id="dataSource.dataSourceTags.tags"
          values={
            Object {
              "tags": 2,
            }
          }
        />
      </styled.div>
    </styled.div>
    <Formik
      enableReinitialize={true}
      initialValues={
        Array [
          "1",
        ]
      }
      isInitialValid={false}
      onSubmit={[Function]}
      validateOnBlur={true}
      validateOnChange={true}
    >
      <Component />
    </Formik>
  </LoadingWrapper>
  <DataSourceNavigation
    dataSource={
      Object {
        "project": Object {
          "id": "1",
        },
      }
    }
    fetchTags={
      [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "dataSourceId": "1",
            },
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": Promise {},
          },
        ],
      }
    }
    hideOnDesktop={true}
    history={
      Object {
        "goBack": [Function],
        "push": [Function],
      }
    }
    intl={
      Object {
        "formatMessage": [Function],
      }
    }
    match={
      Object {
        "params": Object {
          "dataSourceId": "1",
        },
        "url": "/datasource/1/tag",
      }
    }
    setTags={[Function]}
    tags={
      Array [
        Object {
          "id": 1,
          "isActive": true,
          "key": "key 1",
        },
        Object {
          "id": 2,
          "isActive": false,
          "key": "key 2",
        },
      ]
    }
    userRole="admin"
  />
</Fragment>
`;

exports[`DataSourceTags: Component should render correctly when no tags 1`] = `
<Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
    title="Tags List"
  />
  <MobileMenu
    headerSubtitle={
      <FormattedMessage
        defaultMessage="Tags List"
        id="dataSource.dataSourceTags.subTitle"
        values={Object {}}
      />
    }
    options={
      Array [
        Object {
          "allowedRoles": Array [
            "admin",
            "editor",
          ],
          "id": "projectsNavBtn",
          "label": <FormattedMessage
            defaultMessage="Projects"
            id="project.view.projectsListPage"
            values={Object {}}
          />,
          "to": "/project/",
          "type": "link_item",
        },
        Object {
          "allowedRoles": Array [
            "admin",
          ],
          "id": "usersPageNavBtn",
          "label": <FormattedMessage
            defaultMessage="Users"
            id="project.view.usersPage"
            values={Object {}}
          />,
          "to": "/user/",
          "type": "link_item",
        },
        Object {
          "allowedRoles": Array [
            "admin",
            "editor",
          ],
          "id": "projectDetailsNavBtn",
          "label": <FormattedMessage
            defaultMessage="Project details"
            id="project.view.projectDetails"
            values={Object {}}
          />,
          "to": "/project/1",
          "type": "link_item",
        },
        Object {
          "allowedRoles": Array [
            "admin",
            "editor",
          ],
          "id": "dataSourceNavBtn",
          "label": <FormattedMessage
            defaultMessage="Data Sources"
            id="project.view.projectDataSources"
            values={Object {}}
          />,
          "to": "/project/1/datasource",
          "type": "link_item",
        },
        Object {
          "allowedRoles": Array [
            "admin",
            "editor",
          ],
          "id": "stateNavBtn",
          "label": <FormattedMessage
            defaultMessage="States"
            id="project.view.states"
            values={Object {}}
          />,
          "to": "/project/1/state",
          "type": "link_item",
        },
        Object {
          "allowedRoles": Array [
            "admin",
          ],
          "id": "usersNavBtn",
          "label": <FormattedMessage
            defaultMessage="Project Users"
            id="project.view.projectUsersPage"
            values={Object {}}
          />,
          "to": "/project/1/user",
          "type": "link_item",
        },
        Object {
          "allowedRoles": Array [
            "admin",
            "editor",
          ],
          "id": "folderNavBtn",
          "label": <FormattedMessage
            defaultMessage="Pages"
            id="project.view.projectPages"
            values={Object {}}
          />,
          "to": "/project/1/folder",
          "type": "link_item",
        },
      ]
    }
  />
  <ContextHeader
    subtitle={
      <FormattedMessage
        defaultMessage="Tags List"
        id="dataSource.dataSourceTags.subTitle"
        values={Object {}}
      />
    }
  >
    <DataSourceNavigation
      dataSource={
        Object {
          "project": Object {
            "id": "1",
          },
        }
      }
      fetchTags={
        [MockFunction] {
          "calls": Array [
            Array [
              Object {
                "dataSourceId": "1",
              },
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": Promise {},
            },
          ],
        }
      }
      filters={Array []}
      hideOnDesktop={false}
      history={
        Object {
          "goBack": [Function],
          "push": [Function],
        }
      }
      intl={
        Object {
          "formatMessage": [Function],
        }
      }
      match={
        Object {
          "params": Object {
            "dataSourceId": "1",
          },
          "url": "/datasource/1/tag",
        }
      }
      setTags={[Function]}
      tags={
        Array [
          Object {
            "id": 1,
            "isActive": true,
            "key": "key 1",
          },
          Object {
            "id": 2,
            "isActive": false,
            "key": "key 2",
          },
        ]
      }
      userRole="admin"
    />
  </ContextHeader>
  <LoadingWrapper
    error={null}
    loading={false}
    noData={false}
    noDataContent={null}
  >
    <styled.div>
      <styled.div>
        <Styled(Component)
          onClick={[Function]}
        >
          <Component />
        </Styled(Component)>
      </styled.div>
      <styled.div>
        <FormattedMessage
          defaultMessage="{tags, plural,one {# Tag List} other {# Tags List}}"
          id="dataSource.dataSourceTags.tags"
          values={
            Object {
              "tags": 2,
            }
          }
        />
      </styled.div>
    </styled.div>
    <Formik
      enableReinitialize={true}
      initialValues={
        Array [
          "1",
        ]
      }
      isInitialValid={false}
      onSubmit={[Function]}
      validateOnBlur={true}
      validateOnChange={true}
    >
      <Component />
    </Formik>
  </LoadingWrapper>
  <DataSourceNavigation
    dataSource={
      Object {
        "project": Object {
          "id": "1",
        },
      }
    }
    fetchTags={
      [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "dataSourceId": "1",
            },
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": Promise {},
          },
        ],
      }
    }
    filters={Array []}
    hideOnDesktop={true}
    history={
      Object {
        "goBack": [Function],
        "push": [Function],
      }
    }
    intl={
      Object {
        "formatMessage": [Function],
      }
    }
    match={
      Object {
        "params": Object {
          "dataSourceId": "1",
        },
        "url": "/datasource/1/tag",
      }
    }
    setTags={[Function]}
    tags={
      Array [
        Object {
          "id": 1,
          "isActive": true,
          "key": "key 1",
        },
        Object {
          "id": 2,
          "isActive": false,
          "key": "key 2",
        },
      ]
    }
    userRole="admin"
  />
</Fragment>
`;

exports[`DataSourceTags: Component should render correctly with loader 1`] = `
<Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
    title="Tags List"
  />
  <MobileMenu
    headerSubtitle={
      <FormattedMessage
        defaultMessage="Tags List"
        id="dataSource.dataSourceTags.subTitle"
        values={Object {}}
      />
    }
    options={
      Array [
        Object {
          "allowedRoles": Array [
            "admin",
            "editor",
          ],
          "id": "projectsNavBtn",
          "label": <FormattedMessage
            defaultMessage="Projects"
            id="project.view.projectsListPage"
            values={Object {}}
          />,
          "to": "/project/",
          "type": "link_item",
        },
        Object {
          "allowedRoles": Array [
            "admin",
          ],
          "id": "usersPageNavBtn",
          "label": <FormattedMessage
            defaultMessage="Users"
            id="project.view.usersPage"
            values={Object {}}
          />,
          "to": "/user/",
          "type": "link_item",
        },
        Object {
          "allowedRoles": Array [
            "admin",
            "editor",
          ],
          "id": "projectDetailsNavBtn",
          "label": <FormattedMessage
            defaultMessage="Project details"
            id="project.view.projectDetails"
            values={Object {}}
          />,
          "to": "/project/1",
          "type": "link_item",
        },
        Object {
          "allowedRoles": Array [
            "admin",
            "editor",
          ],
          "id": "dataSourceNavBtn",
          "label": <FormattedMessage
            defaultMessage="Data Sources"
            id="project.view.projectDataSources"
            values={Object {}}
          />,
          "to": "/project/1/datasource",
          "type": "link_item",
        },
        Object {
          "allowedRoles": Array [
            "admin",
            "editor",
          ],
          "id": "stateNavBtn",
          "label": <FormattedMessage
            defaultMessage="States"
            id="project.view.states"
            values={Object {}}
          />,
          "to": "/project/1/state",
          "type": "link_item",
        },
        Object {
          "allowedRoles": Array [
            "admin",
          ],
          "id": "usersNavBtn",
          "label": <FormattedMessage
            defaultMessage="Project Users"
            id="project.view.projectUsersPage"
            values={Object {}}
          />,
          "to": "/project/1/user",
          "type": "link_item",
        },
        Object {
          "allowedRoles": Array [
            "admin",
            "editor",
          ],
          "id": "folderNavBtn",
          "label": <FormattedMessage
            defaultMessage="Pages"
            id="project.view.projectPages"
            values={Object {}}
          />,
          "to": "/project/1/folder",
          "type": "link_item",
        },
      ]
    }
  />
  <ContextHeader
    subtitle={
      <FormattedMessage
        defaultMessage="Tags List"
        id="dataSource.dataSourceTags.subTitle"
        values={Object {}}
      />
    }
  >
    <DataSourceNavigation
      dataSource={
        Object {
          "project": Object {
            "id": "1",
          },
        }
      }
      fetchTags={[Function]}
      hideOnDesktop={false}
      history={
        Object {
          "goBack": [Function],
          "push": [Function],
        }
      }
      intl={
        Object {
          "formatMessage": [Function],
        }
      }
      match={
        Object {
          "params": Object {
            "dataSourceId": "1",
          },
          "url": "/datasource/1/tag",
        }
      }
      setTags={[Function]}
      tags={
        Array [
          Object {
            "id": 1,
            "isActive": true,
            "key": "key 1",
          },
          Object {
            "id": 2,
            "isActive": false,
            "key": "key 2",
          },
        ]
      }
      userRole="admin"
    />
  </ContextHeader>
  <LoadingWrapper
    error={null}
    loading={true}
    noData={false}
    noDataContent={null}
  >
    <styled.div>
      <styled.div>
        <Styled(Component)
          onClick={[Function]}
        >
          <Component />
        </Styled(Component)>
      </styled.div>
      <styled.div>
        <FormattedMessage
          defaultMessage="{tags, plural,one {# Tag List} other {# Tags List}}"
          id="dataSource.dataSourceTags.tags"
          values={
            Object {
              "tags": 2,
            }
          }
        />
      </styled.div>
    </styled.div>
    <Formik
      enableReinitialize={true}
      initialValues={
        Array [
          "1",
        ]
      }
      isInitialValid={false}
      onSubmit={[Function]}
      validateOnBlur={true}
      validateOnChange={true}
    >
      <Component />
    </Formik>
  </LoadingWrapper>
  <DataSourceNavigation
    dataSource={
      Object {
        "project": Object {
          "id": "1",
        },
      }
    }
    fetchTags={[Function]}
    hideOnDesktop={true}
    history={
      Object {
        "goBack": [Function],
        "push": [Function],
      }
    }
    intl={
      Object {
        "formatMessage": [Function],
      }
    }
    match={
      Object {
        "params": Object {
          "dataSourceId": "1",
        },
        "url": "/datasource/1/tag",
      }
    }
    setTags={[Function]}
    tags={
      Array [
        Object {
          "id": 1,
          "isActive": true,
          "key": "key 1",
        },
        Object {
          "id": 2,
          "isActive": false,
          "key": "key 2",
        },
      ]
    }
    userRole="admin"
  />
</Fragment>
`;
